### PR TITLE
Add time skip option for offline node

### PR DESCRIPTION
**Pull request by ojura**
Introduces a "skip" option which skips first _t_ seconds.

### CODE REVIEW

**Comment by guilhermelawless** [comment]
@ojura this will also skip transforms in **/tf_static** when using `use_bag_transforms` and skipping time, right? Is this intented / could you consider a special case where those are included?

**Comment by ojura** [comment]
Actually, tfs won’t get skipped. All messages are still processed by the main loop, just dispatching sensor messages is skipped if time < time_first + time_to_skip.

**Comment by ojura** [comment]
@cschuet @gaschler PTAL, I updated this PR to remove the dependency on #661 which was closed without merging.

This is a really handy little option which enables slamming from any part of the bag.

File: cartographer_ros/cartographer_ros/offline_node.cc [hunk 1]
```diff
@@ -60,6 +61,9 @@ DEFINE_bool(load_frozen_state, true,
 DEFINE_bool(keep_running, false,
             "Keep running the offline node after all messages from the bag "
             "have been processed.");
+DEFINE_double(skip, 0,
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/offline_node.cc]: Can we rename this to skip_seconds or anything more descriptive than skip?

File: cartographer_ros/cartographer_ros/offline_node.cc [hunk 2]
```diff
@@ -233,12 +237,24 @@ void RunOfflineNode(const MapBuilderFactory& map_builder_factory) {
...
+    if (!::ros::ok()) {
+      return;
+    }
+
     const auto next_msg_tuple = playable_bag_multiplexer.GetNextMessage();
     const rosbag::MessageInstance& msg = std::get<0>(next_msg_tuple);
     const int bag_index = std::get<1>(next_msg_tuple);
     const bool is_last_message_in_bag = std::get<2>(next_msg_tuple);
 
+    if (!begin_time.has_value()) {
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/offline_node.cc]: Is there no way to peek the first message rather than having to do this check every time?

Comment (ojura) [reply cartographer_ros/cartographer_ros/offline_node.cc]: We can extend `PlayableBagMultiplexer` to allow for peeking the time of the next message, if you prefer.

**Comment by cschuet** [comment]
@ojura Awesome! Let's do it!

File: cartographer_ros/cartographer_ros/offline_node.cc (hunk 2 above)
Comment (cschuet) [reply cartographer_ros/cartographer_ros/offline_node.cc]: @ojura Awesome! Let's do it!

**Comment by ojura** [comment]
@cschuet updated.

**Comment by cschuet** [comment]
@ojura Thanks!

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**